parsers/CA_PE.py

```diff
@@ -1,160 +1,147 @@
 #!/usr/bin/env python3
-
 import json
 from datetime import datetime
 from logging import Logger, getLogger
 from typing import Any
+from zoneinfo import ZoneInfo
 
-# The arrow library is used to handle datetimes consistently with other parsers
-import arrow
 from requests import Session
 
-timezone = "Canada/Atlantic"
-
-
-def _find_pei_key(pei_list, sought_key):
-    matching_item = [
-        item
-        for item in pei_list
-        if "header" in item["data"] and item["data"]["header"].startswith(sought_key)
-    ]
-
-    if not matching_item:
-        return None
-
-    return matching_item[0]["data"]["actualValue"]
-
-
-def _get_pei_info(requests_obj):
-    url = "https://wdf.princeedwardisland.ca/api/workflow"
-    request = {"featureName": "WindEnergy", "queryName": "WindEnergy"}
-    headers = {"Content-Type": "application/json"}
-    response = requests_obj.post(url, data=json.dumps(request), headers=headers)
-
-    raw_data = response.json().get("data") or []
-
-    datetime_item = [
-        item["data"]["text"] for item in raw_data if "text" in item["data"]
-    ]
-    if not datetime_item:
-        # unable to get a timestamp, return empty
-        return None
-    datetime_text = datetime_item[0][len("Last updated ") :]
-    data_timestamp = arrow.get(datetime_text, "MMMM D, YYYY HH:mm A").replace(
-        tzinfo="Canada/Atlantic"
+from electricitymap.contrib.lib.models.event_lists import (
+    ExchangeList,
+    ProductionBreakdownList,
+)
+from electricitymap.contrib.lib.models.events import ProductionMix
+from electricitymap.contrib.lib.types import ZoneKey
+from parsers.lib.exceptions import ParserException
+
+SOURCE = "princeedwardisland.ca"
+TIMEZONE = ZoneInfo("Canada/Atlantic")
+# The dashboard associated with this API can be found at
+# https://www.princeedwardisland.ca/en/feature/pei-wind-energy.
+URL = "https://wdf.princeedwardisland.ca/api/workflow"  # API
+ZONE_KEY = ZoneKey("CA-PE")
+
+
+def _parse_power(text: str) -> float:
+    return float(text[: text.index(" MW")])
+
+
+def _get_event(session: Session):
+    # Request an event from the source.
+    event = session.post(
+        URL,
+        data=json.dumps({"featureName": "WindEnergy", "queryName": "WindEnergy"}),
+        headers={"Content-Type": "application/json"},
+    ).json()["data"]
+    # Collect the production modes and power values into a lookup table.
+    modes = dict(
+        (
+            object_["data"]["header"].split(": ", maxsplit=1)
+            for object_ in event
+            if object_["type"] == "GaugeChart"
+        )
     )
```

Comment:
It would be good if this used dict comprehension instead.

Reply:
If I recall correctly, using a dictionary comprehension would have required duplicate code to separate the key from the value, so I did this instead.